### PR TITLE
Added support for JSR test infrastructure in documentation ruleset

### PR DESCRIPTION
**Pull request by schalkms**

### CODE REVIEW

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicClassSpec.kt [hunk 1]
```diff
@@ -41,44 +37,77 @@ class UndocumentedPublicClassSpec : SubjectSpek<UndocumentedPublicClass>({
 				class Nested
 			}"""
 
+	val nestedPublic = """
+    		/** Some doc */
+			class TestNested {
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicClassSpec.kt]: Spacing look kinda odd here

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicFunctionSpec.kt [hunk 2]
```diff
@@ -1,19 +1,114 @@
...
+    				}
+    			}
+			"""
+			assertThat(subject.compileAndLint(code)).hasSize(2)
+		}
+
+		it("does not report documented public function") {
+			val code = """
+    			/**
+			     * Comment
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicFunctionSpec.kt]: Here too

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicFunctionSpec.kt [hunk 3]
```diff
@@ -1,19 +1,114 @@
...
+    				private fun no2(){}
+    			}
+			"""
+			assertThat(subject.compileAndLint(code)).isEmpty()
+		}
+
+		it("does not report undocumented nested function") {
+			val code = """
+    			/**
+			     * Comment
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicFunctionSpec.kt]: Here too

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicFunctionSpec.kt [hunk 4]
```diff
@@ -1,19 +1,114 @@
...
+					private fun nope4() {}
+				}
+			"""
+			assertThat(subject.compileAndLint(code)).isEmpty()
+		}
+
+		it("does not report public functions in private class") {
+			val code = """
+    			private class NoComments {
+
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicFunctionSpec.kt]: Here too

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/CommentOverPrivateMethodSpec.kt [hunk 5]
```diff
@@ -29,7 +32,18 @@ class CommentOverPrivateMethodSpec : SubjectSpek<CommentOverPrivateFunction>({
...
+			assertThat(subject.compileAndLint(code)).hasSize(0)
+		}
+
+		it("does not report public method in a class with a comment") {
+			val code = """
+    			class Test {
+					/**
+					 * asdf
+					 */
+					fun f() {}
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/CommentOverPrivateMethodSpec.kt]: Here too, or maybe its github?

Comment (schalkms) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/CommentOverPrivateMethodSpec.kt]: I think it's the Github diff viewer.
On my machine it looks *normal* in IntelliJ and SublimeText.

**Pull request merged**